src/macros.jl

```diff
@@ -994,34 +994,29 @@ macro variable(args...)
         anon_singleton = false
     end
 
-    haslb = false
-    hasub = false
-    hasfix = false
-    hasstart = false
-    fixedvalue = NaN
+    info_kwargs = filter(isinfokeyword, kwargs)
+    extra_kwargs = filter(kw -> kw.args[1] != :basename && !isinfokeyword(kw), kwargs)
+    basename_kwargs = filter(kw -> kw.args[1] == :basename, kwargs)
+    infoexpr = VariableInfoExpr(; keywordify.(info_kwargs)...)
     var = x
-    lb = NaN
-    ub = NaN
     # Identify the variable bounds. Five (legal) possibilities are "x >= lb",
     # "x <= ub", "lb <= x <= ub", "x == val", or just plain "x"
     explicit_comparison = false
     if isexpr(x,:comparison) # two-sided
         explicit_comparison = true
-        hasub = true
-        haslb = true
         if x.args[2] == :>= || x.args[2] == :≥
             # ub >= x >= lb
             x.args[4] == :>= || x.args[4] == :≥ || _error("Invalid variable bounds")
             var = x.args[3]
-            lb = esc_nonconstant(x.args[5])
-            ub = esc_nonconstant(x.args[1])
+            setlowerbound_or_error(infoexpr, esc_nonconstant(x.args[5]), _error)
```

Comment:
The `parseXXXconstraint` functions take `_error` as the first argument. Any reason not to be consistent?